### ISSUE TITLE
Desktop Capturer API Failure on CI

### ISSUE THREAD

**Issue by MarshallOfSound** [electron/electron#17832]
```
TypeError: Cannot read property 'resolve' of undefined
    at DesktopCapturer.desktopCapturer.emit (/home/builduser/project/src/out/Default/resources/electron.asar/browser/desktop-capturer.js:48:24) TypeError: Cannot read property 'resolve' of undefined
    at DesktopCapturer.desktopCapturer.emit (/home/builduser/project/src/out/Default/resources/electron.asar/browser/desktop-capturer.js:48:24)
```

Guessing some kind of race condition but will investigate.

Refs: https://circleci.com/gh/electron/electron/175158

**Comment by MarshallOfSound** [comment]
This appears to be an architectural flaw in our desktop capturer API that has existed for _a long time_ (namely a race condition / double callback)

**Comment by MarshallOfSound** [comment]
Fixed by #18029
